katdal/datasources.py

```diff
@@ -17,12 +17,110 @@
 """Various sources of correlator data and metadata."""
 
 import urlparse
+import os
 
 import katsdptelstate
+import redis
 import numpy as np
 
-from .dataset import AttrsSensors
 from .sensordata import TelstateSensorData
+from .chunkstore_rados import RadosChunkStore
+
+
+class DataSourceNotFound(Exception):
+    """File associated with DataSource not found or server not responding."""
+
+
+class DaskLazyIndexer(object):
+    """Turn a dask Array into a LazyIndexer by computing it upon indexing."""
+    def __init__(self, dask_array):
+        self.da = dask_array
+
+    def __getitem__(self, keep):
+        return self.da[keep].compute()
+
```

Comment:
Is this intended to be fully compatible with LazyIndexer? The latter has `__len__`, `__iter__` and a `.name` attribute.

Reply:
Not really yet, although I could add those easily.